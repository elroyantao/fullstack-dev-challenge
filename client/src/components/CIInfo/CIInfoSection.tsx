import { Box, BoxProps, Text } from '@chakra-ui/react'
import React from 'react'
import theme from '../../theme'

interface Props extends BoxProps {
  amount: String
}

const CIInfoSection = ({ title, amount, ...props }: Props) => {
  return (
    <Box {...props} flex="1" w="100%" p={4} borderBottom={`1px solid ${theme.colors.grey3}`}>
      <Box>
        <Text fontSize="sm">{title}</Text>
      </Box>
      <Box>
        <Text fontSize="31px" fontWeight="bold">{amount}</Text>
      </Box>
    </Box>
  )
}

export default CIInfoSection
