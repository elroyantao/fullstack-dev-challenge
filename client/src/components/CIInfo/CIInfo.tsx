import { Box, Flex, Text } from '@chakra-ui/react'
import React from 'react'
import { IProjection } from '../../api/compoundInterest/calculateProjection'
import CIInfoSection from './CIInfoSection'

type Props = {
    projections: IProjection[]
}

const CIInfo: React.FC<Props> = ({ projections }: Props) => {
    if (!projections.length) {
        return (
            <Box width="100%" bg="blue400" height="100%" borderRadius="8px" p={8}>
                <Text fontSize="xl">
                    Use our compound interest calculator to see how much your savings or investments
                    might grow over time. You can include regular deposits or withdrawals.
                </Text>
            </Box>
        )
    }

    const { totalDeposit, totalInterest, balance } = projections[projections.length - 1]

    return (
        <Flex direction="column" h="100%">
            <Box m={-4} mb={0} p={4} bg="primary" color="white" fontWeight="bold">
                Your Savings Details
            </Box>
            <Flex
                direction="column"
                align="flex-start"
                justify="space-around"
                h="100%"
                m={-4}
                mt={0}
            >
                <CIInfoSection title="Total Deposits" amount={totalDeposit} borderBottom="1px " />
                <CIInfoSection title="Total Interest Paid" amount={totalInterest} />
                <CIInfoSection title="Your Balance" amount={balance} />
            </Flex>
        </Flex>
    )
}

export default CIInfo
