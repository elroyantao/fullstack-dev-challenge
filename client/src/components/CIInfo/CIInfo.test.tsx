import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import CIInfo from './CIInfo'

const mockedProjection = [
    {
        month: 0,
        deposit: 1000,
        totalDeposit: 1000,
        totalInterest: 0,
        interest: 0,
        balance: 1000,
    },
    {
        month: 1,
        deposit: 100,
        totalDeposit: 1100,
        totalInterest: 16.67,
        interest: 16.67,
        balance: 1116.67,
    },
]

describe('<CIInfo />', () => {
    describe('@renders', () => {
        it('with projection', () => {
            const wrapper = shallow(<CIInfo projections={mockedProjection} />)
            expect(toJson(wrapper)).toMatchSnapshot()
        })

        it('with projection', () => {
            const wrapper = shallow(<CIInfo projections={[]} />)
            expect(toJson(wrapper)).toMatchSnapshot()
        })
    })
})
