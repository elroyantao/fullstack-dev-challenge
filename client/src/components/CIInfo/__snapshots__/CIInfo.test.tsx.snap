// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CIInfo /> @renders with projection 1`] = `
<Flex
  direction="column"
  h="100%"
>
  <Box
    bg="primary"
    color="white"
    fontWeight="bold"
    m={-4}
    mb={0}
    p={4}
  >
    Your Savings Details
  </Box>
  <Flex
    align="flex-start"
    direction="column"
    h="100%"
    justify="space-around"
    m={-4}
    mt={0}
  >
    <CIInfoSection
      amount={1100}
      borderBottom="1px "
      title="Total Deposits"
    />
    <CIInfoSection
      amount={16.67}
      title="Total Interest Paid"
    />
    <CIInfoSection
      amount={1116.67}
      title="Your Balance"
    />
  </Flex>
</Flex>
`;

exports[`<CIInfo /> @renders with projection 2`] = `
<Flex
  alignItems="center"
  bg="blue400"
  borderRadius="8px"
  height="100%"
  p={8}
  width="100%"
>
  <Text
    fontSize="xl"
  >
    Use our compound interest calculator to see how much your savings or investments might grow over time. You can include regular deposits or withdrawals.
  </Text>
</Flex>
`;
